### PR TITLE
Scaffolder: Add page displaying available actions

### PR DESCRIPTION
**Pull request by jhaals**
## Hey, I just made a Pull Request!

- Adds a list page for all installed non legacy scaffolder actions with information about their input/output.
- Attach description field to actions (descriptions are sourced from the microsite)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/plain.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ export function createFetchPlainAction(options: {
 
   return createTemplateAction<{ url: string; targetPath?: string }>({
     id: 'fetch:plain',
+    description:
+      "Downloads content and places it in the workspacePath or optionally in a subdirectory specified by the 'targetPath' input option.",
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/plain.ts:32]: 'workspacePath' to keep standard?

**Comment by benjdlambert** [comment]
Should we add some docs about this also?

Also @freben + @rugvip maybe you want to take a glance at this too, as me + johan paired on some of this.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/register.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export function createCatalogRegisterAction(options: {
     | { repoContentsUrl: string; catalogInfoPath?: string }
   >({
     id: 'catalog:register',
+    description: 'Registers entity in the software catalog.',
```
[the remark this answers is not in the document]
Comment (adamdmharvey) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/register.ts:34]: Maybe "from the workspace into the..."?

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/bitbucket.ts [hunk 3]
```diff
@@ -167,6 +167,8 @@ export function createPublishBitbucketAction(options: {
     repoVisibility: 'private' | 'public';
   }>({
     id: 'publish:bitbucket',
+    description:
+      'Initializes a git repository of contents in workspacePath and publishes to Bitbucket.',
```
[the remark this answers is not in the document]
Comment (adamdmharvey) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/bitbucket.ts:171]: for consistency, the azure one was "and publishes **it** to Azure" - should this also be "publishes it to Bitbucket?

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts [hunk 4]
```diff
@@ -43,6 +43,8 @@ export function createPublishGithubAction(options: {
     repoVisibility: 'private' | 'internal' | 'public';
   }>({
     id: 'publish:github',
+    description:
+      'Initializes a git repository of contents in workspacePath and publishes to GitHub.',
```
[the remark this answers is not in the document]
Comment (adamdmharvey) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts:47]: ...publishes **it** to GitHub? (or alternatively, change the azure one to remove "it")

File: plugins/scaffolder/src/api.ts [hunk 5]
```diff
@@ -227,4 +230,13 @@ export class ScaffolderClient implements ScaffolderApi {
       );
     });
   }
+
+  /**
+   * @Returns ListActionsResponse containg all registered actions.
```
Comment (adamdmharvey) [inline plugins/scaffolder/src/api.ts:235]:
```suggestion
   * @returns ListActionsResponse containing all registered actions.
```

File: plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    display: 'inline-block',
+    borderRadius: 5,
+    border: `1px solid ${theme.palette.grey[500]}`,
+    position: 'relative',
+  },
+
+  codeRequired: {
+    '&::after': {
+      position: 'absolute',
+      content: '"*"',
```
Comment (adamdmharvey) [inline plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx:59]: are these quotes correct?

File: plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  if (loading) {
+    return <Progress />;
+  }
+
+  if (error) {
+    return (
+      <ErrorPage
+        statusMessage="Failed to load installed actions"
+        status="500"
+      />
```
Comment (adamdmharvey) [inline plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx:84]: ou this is a new one to me.  different than using `<WarningPanel severity="error">`.  maybe another one to align to https://github.com/backstage/backstage/issues/3781 :)

**Review by adamdmharvey — COMMENTED** [review]
👏🏻 neato

File: plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+  TableHead,
+  TableRow,
+  makeStyles,
+} from '@material-ui/core';
+import { JSONSchema } from '@backstage/catalog-model';
+import { JSONSchema7Definition } from 'json-schema';
+
+const useStyles = makeStyles(theme => ({
+  code: {
+    fontFamily: 'Menlo, monospaced',
```
[the remark this answers is not in the document]
Comment (benjdlambert) [reply plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx:44]: woops. I thought the font was called `monospaced` derp.

File: plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx (hunk 6 above)
Comment (jhaals) [reply plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx:59]: They're correct, thanks for asking!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

File: plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx (hunk 7 above)
Comment (jhaals) [reply plugins/scaffolder/src/components/ActionsPage/ActionsPage.tsx:84]: Good to see an RFC for this. I believe the ErrorPage make sense when the initial content can't be displayed.

**Comment by jhaals** [comment]
Regarding workspace path in the descriptions. Should we name it 'workspacePath' to indicate that it's a variable or simply workspace?

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Comment by jhaals** [comment]
I believe this should be ready for another go of review/merge.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Review by freben — APPROVED** [review]

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4869-tedme932bnlpvog2ds7zw4z7hhiealt6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6046251ef285af31a1a35624

**Pull request merged**